src/net.cpp

```diff
@@ -2577,6 +2589,30 @@ bool CConnman::DisconnectNode(NodeId id)
     return false;
 }
 
+bool CConnman::SetWhitelisted(NodeId id, bool fWhitelisted)
+{
+    LOCK(cs_vNodes);
+    for (CNode* pnode : vNodes) {
+        if (id == pnode->id) {
+            pnode->fWhitelisted = fWhitelisted;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool CConnman::SetManualConnection(NodeId id, bool manual)
+{
+    LOCK(cs_vNodes);
+    for (CNode* pnode : vNodes) {
+        if (id == pnode->id) {
+            pnode->m_manual_connection = manual;
+            return true;
+        }
+    }
+    return false;
+}
+
```

Comment:
Could you maybe make an (optionally inline) method `find_node_by_id` that the above methods use? Looks a bit copy-pastey.

Reply:
refactored out